src/components/select/__tests__/select.test.tsx

```diff
@@ -1,16 +1,70 @@
-import { render } from '@testing-library/react'
+import { render, fireEvent } from '@testing-library/react'
 import { Select } from '../Select'
+import '@testing-library/jest-dom'
 
 describe('Select', () => {
   it('should render correctly', () => {
-    const { container } = render(
+    const { container } = render(<Select />)
+    expect(container).toMatchSnapshot()
+  })
+
+  it('should call onSearch when search input changes', () => {
+    const handleSearch = jest.fn()
+    const { getByRole } = render(
       <Select
+        showSearch
+        onSearch={handleSearch}
         options={[
           { label: 'UHC', value: 'united_healthcare' },
           { label: 'Aetna', value: 'aetna' },
         ]}
       />
     )
-    expect(container).toMatchSnapshot()
+    const input = getByRole('combobox')
+    fireEvent.focus(input)
+    fireEvent.change(input, { target: { value: 'UHC' } })
+    expect(handleSearch).toHaveBeenCalledWith('UHC')
+  })
+
+  it('should show MagnifyingGlass icon when focused and showSearch is true', async () => {
```

Comment:
Could we add some snapshot tests for these other examples as well as the base case? I find the component updates to be pretty trivial, and the diff helps specify what the exact change was in relation to different behaviors